### DISCUSSION TITLE
Unable to coerce string to number?

### DISCUSSION THREAD

**Discussion by joel-garcia** [colinhacks/zod#3241] [Q&A] [answered]
I am very new to zod, so forgive any ignorance. I am trying to coerce a string to a number in order to validate it for insertion via prisma into a mongodb collection.

I have tried several methods of coercion, and it seems the most basic method of z.coerce.number() should be enough for my situation, but all attempts result in a prisma error which states:
"Argument `viewCount`: Invalid value provided. Expected Int, provided String."

I'm assuming this means that the coercion has somehow failed and resulted in a string instead of the expected number. My test case is using the following value and I am unclear on why the coerce would have failed:
viewCount: "750712690"

Any input is appreciated.

**Comment by JacobWeisenburger** [comment]
This seems to be working as expected. Please let me know if there is something I am missing.
```ts
console.log( z.coerce.number().parse( '750712690' ) ) // 750712690
```

**Reply by joel-garcia** [reply]
I'm sure it is something I am missing, not you. When I console.log your example, it processes, of course. But how do I make use of that coerce feature from within the definition of my ZodObject which I use for validation? Am I even asking that correctly?

If I do the following:
```ts
// myDummyData and the testSchema have additional data fields which are parsing correctly.
// I used the same z.coerce.number().parse() that worked in the console.log to try and confirm it worked the same way here. 

const testSchema = z.object({
    count: z.coerce.number().parse('750712690')
});

const validation = testSchema.safeParse(myDummyData) ;

if (!validation.success)
    return NextResponse.json(validation.error.errors, { status: 400 });
```
This gives me the error:
⨯ TypeError: keyValidator._parse is not a function

So, I assume I am not doing something right. I have used z.coerce.date() on another field to validate a timestamp string into a DateTime field of my schema. So I thought I would be able to do the same to validate another field which contains a number as a string into a Int field of my schema.

**Comment by JacobWeisenburger** [comment, marked as answer]
Try this:

```ts
const testSchema = z.object( {
    count: z.coerce.number(),
} )

const myDummyData = {
    count: '750712690',
}

const validation = testSchema.safeParse( myDummyData )

if ( validation.success ) {
    // do something with validation.data
    console.log( validation.data )
} else {
    // do something with validation.error.errors
    console.log( validation.error.errors )
}
```

If you found my answer satisfactory, please consider supporting me. Even a small amount is greatly appreciated. Thanks friend! 🙏
https://github.com/sponsors/JacobWeisenburger

**Reply by joel-garcia** [reply]
Thanks for your time, Jacob. In the end, my issue was inexperience. I was setting it up correctly from the start, but after validation I was trying to send the original data to the schema instead of validation.data which meant I had successfully coerced the data and then promptly ignored it. I have it working now by passing validation.data into prisma create.
